graphics/fonts/macfont.cpp

```diff
@@ -591,36 +594,15 @@ MacFONTFont *MacFONTFont::scaleFont(const MacFONTFont *src, int newSize, int sla
 		dododo = i == ccc;
 #endif
 
-		srcSurf.fillRect(Common::Rect(srcSurf.w, srcSurf.h), 0);
-		src->drawChar(&srcSurf, i + src->_data._firstChar, 0, 0, 1);
-		memset(dstGray, 0, dstGraySize * sizeof(int));
-		magnifyGray(&srcSurf, dstGray, srcglyph->width, src->_data._fRectHeight, scale);
+		// srcSurf.fillRect(Common::Rect(srcSurf.w, srcSurf.h), 0);
+		// src->drawChar(&srcSurf, i + src->_data._firstChar, 0, 0, 1);
+		// memset(dstGray, 0, dstGraySize * sizeof(int));
+		// magnifyGray(&srcSurf, dstGray, srcglyph->width, src->_data._fRectHeight, scale);
 
 		MacGlyph *glyph = (i == src->_data._glyphs.size()) ? &data._defaultChar : &data._glyphs[i];
-		int *grayPtr = dstGray;
-
-		for (int y = 0; y < data._fRectHeight; y++) {
-			byte *dst = (byte *)srcSurf.getBasePtr(0, y);
-
-			for (int x = 0; x < glyph->bitmapWidth; x++, grayPtr++, dst++) {
-#if DEBUGSCALING
-				if (i == ccc) {
-					if (*grayPtr)
-						debugN(1, "%3d ", *grayPtr);
-					else
-						debugN(1, "    ");
-				}
-#endif
-				if (*grayPtr > grayLevel)
-					*dst = 1;
-				else
-					*dst = 0;
-			}
-#if DEBUGSCALING
-			if (i == ccc)
-				debug(1, "");
-#endif
-		}
+		
+		// src->scaleSingleGlyph(&srcSurf, dstGray, glyph->bitmapWidth, data._fRectHeight, grayLevel);
+		Graphics::Surface *srcSurf = srcCopy->scaleSingleGlyph(glyph->bitmapWidth, data._fRectHeight, grayLevel, i + src->_data._firstChar, 0, src->_data._fRectHeight, srcglyph->width, scale);
 
 		if (slant & kMacFontBold) {
 			memset(dstGray, 0, dstGraySize * sizeof(int));
```

Comment:
Where is dstGray defined now?

Reply:
I have declared starting of the code in the new commit.